### PR TITLE
Feature/fixing annotations

### PR DESCRIPTION
**Pull request by RoyNyaga**
Created a test file each for every class as requested.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require './lib/tic_tac_toe.rb'
+
+RSpec.describe Board do
```
Comment (bolah2009) [inline spec/board_spec.rb:5]: #### `REQUIRED`

Consider using `let` instead, this file have 6 different place `board = Board.new` was defined. Using the below suggestion will avoid defining `board` in each example. Learn more [here](https://github.com/rubocop-hq/rspec-style-guide#let-blocks)

```suggestion
RSpec.describe Board do
  let(:board) { Board.new }
  
  describe ' #initialize_Board ' do
```

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  describe ' #initialize_Board ' do
+    it ' test the correct number on each cell of the grid' do
+      board = Board.new
```
Comment (bolah2009) [inline spec/board_spec.rb]: #### `REQUIRED`

Following the above suggestion, this line is not needed anymore.

File: spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        cond = cell.to_i == (index + 1)
+        break unless cond
+      end
+      expect(cond).to be true
+    end
+  end
+
+  describe '#check space' do
+    it 'returns true when a space is availabe in the board' do
+      board = Board.new
```
Comment (bolah2009) [inline spec/board_spec.rb]: #### `REQUIRED`

Following the above suggestion, this line is not needed anymore.

File: spec/board_spec.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      cond = board.space?
+      expect(cond).to be true
+    end
+
+    it 'returns false when a space is not availabe in the board' do
+      board = Board.new
```
Comment (bolah2009) [inline spec/board_spec.rb]: #### `REQUIRED`

Following the above suggestion, this line is not needed anymore.

File: spec/board_spec.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        board.save_choise_player('X', n + 1)
+      end
+      cond = board.space?
+      expect(cond).to be false
+    end
+  end
+
+  describe ' #check_choise_player' do
+    it ' when a player select a number of the empty grid ' do
+      board = Board.new
```
Comment (bolah2009) [inline spec/board_spec.rb]: #### `REQUIRED`

Following the above suggestion, this line is not needed anymore.

File: spec/board_spec.rb [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+             board.check_choise_player(2) &&
+             board.check_choise_player(3)
+
+      expect(cond).to be false
+    end
+  end
+
+  describe ' #save_choise_player ' do
+    it ' when a mark is saved in the grid of each Player selection' do
+      board = Board.new
```
Comment (bolah2009) [inline spec/board_spec.rb]: #### `REQUIRED`

Following the above suggestion, this line is not needed anymore.

File: spec/board_spec.rb [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+  describe ' #check_choise_player' do
+    it ' when a player select a number of the empty grid ' do
+      board = Board.new
+      cond = board.check_choise_player(1) && board.check_choise_player(2)
+      expect(cond).to be true
+    end
+
+    it ' when a player select a number of the grid with saves' do
+      board = Board.new
```
Comment (bolah2009) [inline spec/board_spec.rb]: #### `REQUIRED`

Following the above suggestion, this line is not needed anymore.

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Hi Team :wave: 

## Good job implementing the changes suggested in PR #2. :+1: 

### A few more required changes before a :+1:  :smile:

- Use [`let blocks`](https://github.com/rubocop-hq/rspec-style-guide#let-blocks) before, see comments below.
  > Instead of using before_each to define a variable for the test cases, use [let](https://github.com/rubocop-hq/rspec-style-guide#let-blocks).
- Add a README file
   > Remember about descriptive README file.

As this is a follow-up project from the **Tic-Tac-Toe (OOP)** project can you paste the link of the project on the readme and here too? :thinking: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Comment by RoyNyaga** [comment]
Thanks for the review, we will get to work immediately

**Review by Oghenebrume50 — APPROVED** [review]
Good job guys this is fine, everything works now and perfectly implemented :smile:

**Pull request merged**